library/L2_HAL/communication/bluetooth/bluetooth_device.hpp

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include <initializer_list>
+
+#include "L1_Peripheral/uart.hpp"
+
+namespace sjsu
+{
+namespace bluetooth
+{
+/// The interface should be implemented by bluetooth devices that typically
+/// utilize the CC245x series of SoC for Bluetooth Low Energy (BLE)
+/// communication.
+///
+/// The CC245x MCUs provide BLE v4.0 which allows for communication with modern
+/// devices such as iOS or Android devices. These devices typically use UART
+/// communicate with the MCU.
+class BluetoothDevice
```

Comment:
This Interface doesn't really define all bluetooth devices. Or even most. Just bluetooth devices that act as wireless UART transmitters or (transparent UART). It may be a good idea to really think about the bluetooth module as such, an sjsu::Uart module. Now if you want to have additional interface methods specifically for UartBluetooth devices, then having this sort of interface is good. But you don't want to duplicate the work. So imagine this:

```C++
class HM10 : public sjsu::Uart, sjsu::UartOverBluetoothDevice
{
 public:
  virtual void Disconnect() const { /* ... */ } // from the sjsu::UartOverBluetoothDevice
};
```

sjsu::UartOverBluetoothDevice is just an example name. You can come up with whatever you think makes sense. But the main point is that it would be helpful to put the HM10 (and other devices) in places of where a UART port would go.

See esp8266.hpp for another example of a class that implements two interfaces